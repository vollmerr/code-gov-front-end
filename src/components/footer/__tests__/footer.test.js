import React from 'react';

import Footer from 'components/footer'

import { shallow, mount } from 'enzyme';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares);

describe('Testing <Footer /> component', () => {
  const props = {
    color: "white",
    dispatch: jest.fn(),
    links: [{name: 'test-link', url: '/test-url'}],
    logos: [{image: '/path/to/image', name: 'test-image', url: 'https://test.com'}],
    router: {
      action: 'POP',
      location: {pathname: '/', search: '', hash: '', state: undefined, key: 'test-key' }
    }
  };
  const store = mockStore(props);
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<Footer store={store} />);
  });
  it('renders Footer', () => {
    expect(wrapper.find('div.footer').exists()).toEqual(true);
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should render dark correctly if not onHomePage', () => {
    PUBLIC_PATH='';
    wrapper = mount(<Footer store={store} />);
    expect(wrapper).toMatchSnapshot();
  })
})