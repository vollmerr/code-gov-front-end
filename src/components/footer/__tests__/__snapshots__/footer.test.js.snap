// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Footer /> component should render correctly 1`] = `
<Connect(Footer)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Footer
    color="dark"
    dispatch={[Function]}
    links={
      Array [
        Object {
          "name": "text-link",
          "url": "/test-url",
        },
      ]
    }
    logos={
      Array [
        Object {
          "image": "/test/path/to/logo.png",
          "name": "Test-name",
          "url": "https://Test.gov/",
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="footer"
      role="contentinfo"
    >
      <div
        className="footer-content"
      >
        <div
          className="icon-row"
        >
          <img
            alt="Code California logo"
            className="foot-img"
            src="../../assets/img/logo-reverse.png"
          />
          <div
            className="separator"
          />
          <div>
            <a
              aria-label="Subscribe to the Newsletter"
              className="icon-link"
              href="http://eepurl.com/dDrRC1"
            >
              <i
                aria-hidden="true"
                className="fa fa-envelope"
              />
            </a>
            <a
              aria-label="Link to Medium"
              className="icon-link"
              href="https://medium.com/code-california"
            >
              <i
                aria-hidden="true"
                className="fa fa-medium"
              />
            </a>
            <a
              aria-label="Join us on Slack"
              className="icon-link"
              href="https://join.slack.com/t/codecagov/shared_invite/enQtNDU3MDY2NDUyMTk3LTkxMzEwYzU3MzdkNTE3ZGE0OGY4MjU5YzUzMTE5ODgwZjc3ZDE3MjlhMjQ3NTJkN2Y2NzFmZDk0NzZmNzgwYmU"
            >
              <i
                aria-hidden="true"
                className="fa fa-slack"
              />
            </a>
            <a
              aria-label="Twitter Link"
              className="icon-link"
              href="https://twitter.com/CodeCAgov"
            >
              <i
                aria-hidden="true"
                className="fa fa-twitter"
              />
            </a>
            <a
              aria-label="Facebook Link"
              className="icon-link"
              href="https://www.facebook.com/codeCAgov"
            >
              <i
                aria-hidden="true"
                className="fa fa-facebook"
              />
            </a>
            <a
              aria-label="GitHub Link"
              className="icon-link"
              href="https://github.com/cagov/codeCAgov"
            >
              <i
                aria-hidden="true"
                className="fa fa-github"
              />
            </a>
            <a
              aria-label="Eventbrite Link"
              className="icon-link"
              href="http://codeCAgov.eventbrite.com"
            >
              <i
                aria-hidden="true"
                className="fa fa-calendar"
              />
            </a>
            <a
              aria-label="Medium feed Link"
              className="icon-link"
              href="https://medium.com/feed/code-california"
            >
              <i
                aria-hidden="true"
                className="fa fa-feed"
              />
            </a>
          </div>
          <div
            className="icon-link"
          >
            2019 Code California
          </div>
        </div>
      </div>
    </div>
  </Footer>
</Connect(Footer)>
`;

exports[`Testing <Footer /> component should render dark correctly if not onHomePage 1`] = `
<Connect(Footer)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Footer
    color="white"
    dispatch={[Function]}
    links={
      Array [
        Object {
          "name": "text-link",
          "url": "/test-url",
        },
      ]
    }
    logos={
      Array [
        Object {
          "image": "/test/path/to/logo.png",
          "name": "Test-name",
          "url": "https://Test.gov/",
        },
      ]
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="footer"
      role="contentinfo"
    >
      <div
        className="footer-content"
      >
        <div
          className="icon-row"
        >
          <img
            alt="Code California logo"
            className="foot-img"
            src="../../assets/img/logo-reverse.png"
          />
          <div
            className="separator"
          />
          <div>
            <a
              aria-label="Subscribe to the Newsletter"
              className="icon-link"
              href="http://eepurl.com/dDrRC1"
            >
              <i
                aria-hidden="true"
                className="fa fa-envelope"
              />
            </a>
            <a
              aria-label="Link to Medium"
              className="icon-link"
              href="https://medium.com/code-california"
            >
              <i
                aria-hidden="true"
                className="fa fa-medium"
              />
            </a>
            <a
              aria-label="Join us on Slack"
              className="icon-link"
              href="https://join.slack.com/t/codecagov/shared_invite/enQtNDU3MDY2NDUyMTk3LTkxMzEwYzU3MzdkNTE3ZGE0OGY4MjU5YzUzMTE5ODgwZjc3ZDE3MjlhMjQ3NTJkN2Y2NzFmZDk0NzZmNzgwYmU"
            >
              <i
                aria-hidden="true"
                className="fa fa-slack"
              />
            </a>
            <a
              aria-label="Twitter Link"
              className="icon-link"
              href="https://twitter.com/CodeCAgov"
            >
              <i
                aria-hidden="true"
                className="fa fa-twitter"
              />
            </a>
            <a
              aria-label="Facebook Link"
              className="icon-link"
              href="https://www.facebook.com/codeCAgov"
            >
              <i
                aria-hidden="true"
                className="fa fa-facebook"
              />
            </a>
            <a
              aria-label="GitHub Link"
              className="icon-link"
              href="https://github.com/cagov/codeCAgov"
            >
              <i
                aria-hidden="true"
                className="fa fa-github"
              />
            </a>
            <a
              aria-label="Eventbrite Link"
              className="icon-link"
              href="http://codeCAgov.eventbrite.com"
            >
              <i
                aria-hidden="true"
                className="fa fa-calendar"
              />
            </a>
            <a
              aria-label="Medium feed Link"
              className="icon-link"
              href="https://medium.com/feed/code-california"
            >
              <i
                aria-hidden="true"
                className="fa fa-feed"
              />
            </a>
          </div>
          <div
            className="icon-link"
          >
            2019 Code California
          </div>
        </div>
      </div>
    </div>
  </Footer>
</Connect(Footer)>
`;
